pkg/apis/crd.nsx.vmware.com/v1alpha1/ipblockdata_types.go

```diff
@@ -0,0 +1,53 @@
+/* Copyright © 2024 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type IPBlockVisibility string
+
+const (
+	IPBlockVisibilityExternal   = "External"
+	IPBlockVisibilityPrivate    = "Private"
+	IPBlockVisibilityPrivateTGW = "PrivateTGW"
+)
+
+// IPBlockDataSpec defines the desired state of IPBlockData
+type IPBlockDataSpec struct {
+	// Visibility specifies the visibility or scope of IpAddressBlock
+	// and expected consumption of IpAddressBlock with same scope.
+	// Can be External, Private or PrivateTGW.
+	// +kubebuilder:validation:Enum=External;Private;PrivateTGW
+	// +kubebuilder:default=Private
+	Visibility IPBlockVisibility `json:"visibility,omitempty"`
+	// CIDR is a string representing the IP Block.
```

Comment:
Let us copy NSX API spec: "A contiguous IP address space represented by network address and prefix length"

Reply:
Updated, thanks!